spswarehouse/powerschool/powerschool.py

```diff
@@ -0,0 +1,437 @@
+
+        logging.info("Successful login confirmed!")
+    except:
+        raise Exception("Unable to confirm successful login to PowerSchool. Please check your credentials.")
+
+def get_current_domain(driver: WebDriver):
+    """
+    Retrieves the current domain.
+    
+    Parameters:
+    driver: The Selenium WebDriver
+
+    Returns:
+    str: The current domain
+    """
+
+    return driver.current_url[8:].split("/",1)[:1][0]
+
+def get_current_path(driver: WebDriver):
+    """
+    Retrieves the current path, minus the 'https://'.
+    
+    Parameters:
+    driver: The Selenium WebDriver
+
+    Returns:
+    str: The current path
+    """
+
+    return driver.current_url[8:].split("/",1)[1:][0]
+
+def ensure_on_desired_path(driver: WebDriver, desired_path: str):
+    """
+    Checks whether the WebDriver is on the desired path. If not, navigates there.
+
+    Parameters:
+    driver: The Selenium WebDriver
+    desired_path: The path to be checked against
+
+    Returns:
+    n/a
+    """
+
+    current_path = get_current_path(driver)
+
+    logging.info(f"The current path is: {current_path}")
+
+    if(current_path == desired_path):
+        logging.info("The current path is the desired path. No action taken.")
+    else:
+        logging.info(f"This does not match {desired_path}, so going to that path")
+        driver.get('https://' + get_current_domain(driver) + "/" + desired_path)
+        logging.info(f"Moved to {desired_path}")
+
+def check_whether_desired_school_selected(driver, school_name: str) -> bool:
+    """
+    Checks whether the specified school is currently selected in PowerSchool but
+    takes no action either way.
+
+    Parameters:
+    driver: The Selenium WebDriver
+    school_name: The school name for selecting from the drop-down in the upper-right of the user interface.
+        Should be the complete school name to avoid incorrect partial matches.
+
+    Returns:
+    bool: True indicates the desired school is selected, and False indicates it is not.
+    """
+
+    elem = WebDriverWait(driver, 30).until(EC.presence_of_element_located((By.ID, 'school_picker_adminSchoolPicker_toggle_btn')))
+
+    elem.click()
+    time.sleep(1)
+
+    selected_element = WebDriverWait(driver, 30).until(EC.presence_of_element_located((By.CSS_SELECTOR, '.list-item.selectable.selected')))
+
+    school_element_text = selected_element.find_element(By.XPATH, ".//div").text
+
+    result_message = f"Found the {school_name} in the school element text. Match!"
+    outcome = True
+
+    if(school_name not in school_element_text):
+        result_message = f"Did not find {school_name} in the school element text. No match."
+        outcome = False
+
+    logging.info(result_message)
+
+    logging.info(f"Pressing escape to leave dropdown selection.")
+    actions = ActionChains(driver)
+    actions.send_keys(Keys.ESCAPE).perform()
+    
+    return outcome
+
+def switch_to_school(driver: WebDriver, school_name: str):
```

Comment:
I think if we turn this into a PowerSchool class, I'd make this an internal function too.